### PR TITLE
Add ability to skip issuer https check and token nonce verification

### PR DESCRIPTION
**Pull request by agologan**
### Checklist
- [x] I read the [Contribution Guidelines](https://github.com/openid/AppAuth-Android/blob/master/CONTRIBUTING.md)
- [x] I signed the CLA and WG Agreements 
- [x] I ran, updated and added unit tests as necessary.
- [x] I verified the contribution matches existing coding style.
- [x] I updated the documentation if necessary.

### Motivation and Context

Change aims to address issue in #655, #585 and #657 by adding the ability to disable certain steps in ID Token validation.

### CODE REVIEW

**Comment by carstenhag** [comment]
Looks fitting for my use case (#655), thanks for adding this :)

File: library/java/net/openid/appauth/AppAuthConfiguration.java [hunk 1]
```diff
@@ -94,12 +114,33 @@ public Builder setConnectionBuilder(@NonNull ConnectionBuilder connectionBuilder
             return this;
         }
 
+        /**
+         * Disables https validation for the issuer identifier.
```
Comment (carstenhag) [inline library/java/net/openid/appauth/AppAuthConfiguration.java:118]: Perhaps this should be documented/log a big warning that this is very unsafe? That way people don't disable it without being sure it's the right thing they want to do.

File: library/java/net/openid/appauth/AppAuthConfiguration.java [hunk 2]
```diff
@@ -94,12 +114,33 @@ public Builder setConnectionBuilder(@NonNull ConnectionBuilder connectionBuilder
...
+         */
+        public Builder setSkipIssuerHttpsCheck(Boolean skipIssuerHttpsCheck) {
+            mSkipIssuerHttpsCheck = skipIssuerHttpsCheck;
+            return this;
+        }
+
+        /**
+         * Disables nonce verification for value sent in the Authentication Request.
```
Comment (carstenhag) [inline library/java/net/openid/appauth/AppAuthConfiguration.java:126]: Could also link to a reasoning on why this *should* be enabled, even if it's not the most critical thing.

**Comment by carstenhag** [comment]
Looks good, thanks! :)

**Comment by cumcum8197** [comment]
429b53256f2dce2288a7a1da970cff84dde6099c

**Pull request merged**